### DISCUSSION TITLE
Adding CVXPY abs to optimization problem turns out to be non-DCP

### DISCUSSION THREAD

**Discussion by sasin324** [cvxpy/cvxpy#1642] [General]
I have tried to solve an optimization problem using CVXPY library. Basically, this problem is regression with additional regularizer, where `sv` is n-dimensional vector containing additional information used to define this regularizer. This problem aims to minimize the loss function and distance between `beta`, which can be positive or negative real numbers, and `sv` containing only positive real number, as follows 

```python
beta = cp.Variable(n)
s = cp.Parameter(len(n), nonneg=False, value=sv)
problem = cp.Problem(cp.Minimize((cp.sum_squares(X @ beta - Y)) + (lambda1 * cp.abs(cp.norm1(beta) - cp.norm1(s)))))
```

I tried `cp.norm1(beta) - cp.norm1(s)` and `cp.norm1(beta - s)` and they worked fine, but I want to make sure that the result of the difference between both vectors must be positive. In other words, I want to consider only the magnitude of beta without its sign in the additional regularizer. However, when I add `abs` in both formulation, I got the error saying that "this problem  does not follow DCP rules". I am not sure how to resolve this error.

**I should have stated my problem more clearly, I am so sorry about that.**
example of 4 dimensional `sv` could be [0.23, 1.05, 3.44, 0.13]

**Comment by keithbriggs** [comment]
Why isn't the answer just `beta=s`?

**Comment by rileyjmurray** [comment]
Is ``s`` a Variable? It would help if you posted a minimal working example.

**Comment by sasin324** [comment]
I have updated my question accordingly. Sorry about the first version

**Reply by rileyjmurray** [reply]
Hi, @sasin324, no worries. So it turns out it doesn't matter whether or not ``s`` was a variable. The term ``cp.abs(cp.norm1(beta) - cp.norm1(s))`` is not convex in ``beta`` even when ``s`` is fixed.

Suppose the 1-norm of ``s`` is one, and define ``f(beta) = abs(norm1(beta) - 1)``. If ``beta0`` has norm 1, then ``f(beta0) = f(-beta0) = 0``, and yet ``f((beta0 + (-beta))/2) = f(0) = 1``, so the requirement for convexity (that ``f(average) <= average(f)``) fails.

**Reply by sasin324** [reply]
Thank you for pointing this out. I understood now why my problem is not convex.
Is there any way that can be used to solve a non-convex problem?

**Reply by rileyjmurray** [reply]
Sadly, there isn't an actively maintained variant of cvxpy which supports this kind of nonconvexity.
